dpiprobe.go

```diff
@@ -58,7 +59,10 @@ func main() {
 		os.Exit(1)
 	}
 
-	encodedDomain, err := idna.ToASCII(domain)
+	if *serverName == "" {
+		*serverName = domain
+	}
+	encodedDomain, err := idna.ToASCII(*serverName)
```

Comment:
You need to make sure that you are connecting to the target IP of the domain not the server name.... We want to probe domain but use server name in the SNI and host header.